packages/web/src/hooks/swap/data/use-swap.tsx

```diff
@@ -115,6 +138,8 @@ export const useSwap = ({ tokenA, tokenB, direction, slippage, swapFee = 15 }: U
   }, [direction, estimatedAmount, slippage, tokenA]);
 
   const updateSwapAmount = useCallback((amount: string) => {
+    if (!amount) return setSwapAmount(null);
+
     let newAmount = 0;
     if (!amount || BigNumber(amount).isZero()) {
```

Comment:
The `!amount` check in here is redundant since the first `if (!amount)` statement (line 141) already handles this case with an early return.

Reply:
Removed unnecessary conditional checks.